demos/signature/EVP_ED_Signature_demo.c

```diff
@@ -0,0 +1,210 @@
+    unsigned char *sig_value = NULL;
+    EVP_MD_CTX *sign_context = NULL;
+
+    /* Create a signature context */
+    sign_context = EVP_MD_CTX_new();
+    if (sign_context == NULL) {
+        fprintf(stderr, "EVP_MD_CTX_new failed.\n");
+        goto cleanup;
+    }
+
+    /*
+     * Initialize the sign context using a ED25519 private key
+     * Notice that the digest name must NOT be used.
+     * In this demo we don't specify any additional parameters via
+     * OSSL_PARAM, which means it will use default values.
+     * For more information, refer to
+     * doc/man7/EVP_SIGNATURE-ED25519.pod
+     * "ED25519 and ED448 Signature Parameters"
+     */
+    if (!EVP_DigestSignInit_ex(sign_context, NULL, NULL, libctx, NULL, priv, NULL)) {
+        fprintf(stderr, "EVP_DigestSignInit_ex failed.\n");
+        goto cleanup;
+    }
+
+    /* Calculate the required size for the signature by passing a NULL buffer. */
+    if (!EVP_DigestSign(sign_context, NULL, &sig_len, tbs, tbs_len)) {
+        fprintf(stderr, "EVP_DigestSign using NULL buffer failed.\n");
+        goto cleanup;
+    }
+    sig_value = OPENSSL_zalloc(sig_len);
+    if (sig_value == NULL) {
+        fprintf(stderr, "No memory.\n");
+        goto cleanup;
+    }
+    fprintf(stdout, "Generating signature:\n");
+    if (!EVP_DigestSign(sign_context, sig_value, &sig_len, tbs, tbs_len)) {
+        fprintf(stderr, "EVP_DigestSign failed.\n");
+        goto cleanup;
+    }
+    *sig_out_len = sig_len;
+    *sig_out_value = sig_value;
+    BIO_dump_indent_fp(stdout, sig_value, sig_len, 2);
+    fprintf(stdout, "\n");
+    ret = 1;
+
+cleanup:
+    /* OpenSSL free functions will ignore NULL arguments */
+    if (!ret)
+        OPENSSL_free(sig_value);
+    EVP_MD_CTX_free(sign_context);
+    return ret;
+}
+
+static int demo_verify(EVP_PKEY *pub,
+                       const unsigned char *tbs, size_t tbs_len,
+                       const unsigned char *sig_value, size_t sig_len,
+                       OSSL_LIB_CTX *libctx)
+{
+    int ret = 0;
+    EVP_MD_CTX *verify_context = NULL;
+
+    /*
+     * Make a verify signature context to hold temporary state
+     * during signature verification
+     */
+    verify_context = EVP_MD_CTX_new();
+    if (verify_context == NULL) {
+        fprintf(stderr, "EVP_MD_CTX_new failed.\n");
+        goto cleanup;
+    }
+    /* Initialize the verify context with a ED25519 public key */
+    if (!EVP_DigestVerifyInit_ex(verify_context, NULL, NULL,
+                                 libctx, NULL, pub, NULL)) {
+        fprintf(stderr, "EVP_DigestVerifyInit_ex failed.\n");
+        goto cleanup;
+    }
+    /*
+     * ED25519 only supports the one shot interface using EVP_DigestVerify()
+     * The streaming EVP_DigestVerifyUpdate() API is not supported.
+     */
+    if (!EVP_DigestVerify(verify_context, sig_value, sig_len,
+                          tbs, tbs_len)) {
+        fprintf(stderr, "EVP_DigestVerifyUpdate(hamlet_1) failed.\n");
```

Comment:
Replace the error text with `EVP_DigestVerify failed.\n`

Reply:
thanks